packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte

```diff
@@ -0,0 +1,383 @@
+<script lang="ts">
+import type { Registry } from '@tmpwip/extension-api';
+import { onMount } from 'svelte';
+import { registriesInfos } from '../../stores/registries';
+import DropdownMenu from '../ui/DropdownMenu.svelte';
+import DropdownMenuItem from '../ui/DropDownMenuItem.svelte';
+import { faTrash } from '@fortawesome/free-solid-svg-icons';
+import { faUser } from '@fortawesome/free-solid-svg-icons';
+import { faUserPen } from '@fortawesome/free-solid-svg-icons';
+
+let modifiedRegistries: Registry[] = [];
+let loginErrorMessages: { serverUrl: string; error: string }[] = [];
+let showPasswordForServerUrls: string[] = [];
+let showNewRegistryForm = false;
+let providerSourceName: string;
+const newRegistryRequest = {
+  source: '',
+  serverUrl: '',
+  username: '',
+  secret: '',
+} as Registry;
+
+onMount(async () => {
+  let providerSourceNames = await window.getImageRegistryProviderNames();
+  if (providerSourceNames.length > 0) {
+    providerSourceName = providerSourceNames[0];
+  }
+});
+
+function isRegistryModified(serverUrl: string): boolean {
+  return modifiedRegistries.some(r => r.serverUrl === serverUrl);
+}
+
+function markRegistryAsModified(registry: Registry) {
+  setPasswordForRegistryVisible(registry.serverUrl, false);
+
+  const modifiedRegistry = {
+    source: registry.source,
+    serverUrl: registry.serverUrl,
+    username: registry.username,
+    secret: registry.secret,
+  } as Registry;
+
+  modifiedRegistries.push(modifiedRegistry);
+
+  updateUI();
```

Comment:
hello, why do we have these updateUI() calls ?

UI should refresh automatically

Reply:
it should, but it actually doesn't refresh when we iterate over storage elements and need dynamic change the dom elements on the UI after processing onClick events, updating storage with self elements solves this problem and I think it's like a bug in svelte. so yes, here is a little workaround

Reply:
I think we're not doing something correct on our side.

For example for svelte, we need to reassign a value, not append to an array for example

Reply:
I guess you should do
```
 modifiedRegistries = [...modifiedRegistries, modifiedRegistry]
```

https://svelte.dev/tutorial/updating-arrays-and-objects

Reply:
updated, thanks for pointing to this doc